### PR TITLE
Add example tag provider

### PR DESCRIPTION
**Pull request by pipo02mix**
## Checklist

- [x] Consider SIG UX feedback.
- [ ] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Comment by pipo02mix** [comment]
@marians is this file auto-generated? or I can add it manually?

**Comment by marians** [comment]
Can we make the example all lowercase?

I see no additional value in encouraging users to use mixed case labels. However I see a potential of getting confused when users try to use mixed case labels in selectors and don't know about their case sensitivity.

**Comment by pipo02mix** [comment]
It is copied from an actual customer case. It is up to them choosing the format but I am fine with not encourgint them to use uppercase. I change it

File: docs/cr/cluster.x-k8s.io_v1alpha2_cluster.yaml [hunk 1]
```diff
@@ -3,7 +3,8 @@ kind: Cluster
 metadata:
   annotations:
     giantswarm.io/docs: https://pkg.go.dev/sigs.k8s.io/cluster-api/api/v1alpha2?tab=doc#Cluster
-  creationTimestamp: null
+  labels:
+    tag.provider.giantswarm.io/Environment: PROD
```
Comment (pipo02mix) [inline docs/cr/cluster.x-k8s.io_v1alpha2_cluster.yaml:7]:
```suggestion
    tag.provider.giantswarm.io/environment: prod
```

**Review by marians — APPROVED** [review]
Thanks!

**Pull request merged**